### ISSUE TITLE
Start the application when the user logs on

### ISSUE THREAD

**Issue by ghost** [akej74/grid-control#22]
Hi there,

Can you add a "Start at log on" check box that, once checked, will launch the application when the user logs into windows. Also, to make it even better, add a "Delay start by:" check box under the 1st check box I mentioned (the "Delay start by:" box will be greyed out if the 1st check box is unchecked).  This 2nd check box will have a drop down menu that will allow the user to choose the delay time in seconds in increments of 5 seconds, starting with the value of 5 seconds.

The reason I'm requesting this feature is that windows 7's way of implementing it is flawed (and I prefer not to download another 3rd party app to get it done). I tried creating a shortcut to the executable file and putting that shortcut in the startup folder. However, while this does cause your application to start at log on, it starts it a few seconds before the  Open Hardware Monitor app start and therefore returns an error regarding its inability to get readings. I also tried using Windows 7 Task Scheduler, but the minimum delay time it offers is 30 seconds, which, I think, is too long. Finally, I tried solving this problem by creating a batch file and putting it in the startup folder, but that effort proved fruitless as well.

Hopefully, you'll be able to help solve this problem

Thanks!!!

**Comment by akej74** [comment]
Hi, would something like the following work...? 

1. Grid Control start, check if OHM is running.
 
2. If OHM is not running, just wait e.g 1s and then try again. This is done silently without any popup but with some kind of warning message displayed that Grid Control is waiting for OHM to start.

This would allow Grid Control to be in a "paused" mode waiting for OHM to be available.

**Comment by ghost** [comment]
Yes, this would definitely work. Except, there is no need for the warning message. If OHM doesn't start after 5-10 "1s" attempts, then display some kind of "failed to start" message, otherwise there is no need to inform the user that Grid Control is in waiting mode.

**Comment by akej74** [comment]
New release is up, 1.0.7! I decided to show a message to the user that OHM needs to be started, but it's "non modal" and will close automatically when OHM is up. A count down timer is also displayed just for some excitement to the user as Grid Control will self destruct in 30s in case OHM is not started :-)

Please try this out and see how it works for you.

**Comment by ghost** [comment]
Well, now instead of creating a task schedule for the application I can just place its shortcut in the Startup folder. Then, at log in, I get your message and after about 5 seconds the application will indeed start and the message disappears. So I can confirm your change. This is a good start, but if you could add 2 more things it would be perfect:

1) Add a "Run On System Startup" check box that, once checked, will launch the application when the user logs into windows. The Open Hardware Monitor application has that same option under its Option menu and, once checked, it simply adds a task for the application in Windows's Task Scheduler. This will save the user the need to add the task himself/herself

2) Make that message optional. I prefer a silent startup with no messages appearing on my screen, but I'm sure others would appreciate the option being there. Also, change the message from "Please launch Open Hardware Monitor ..." to "Waiting on "Open Hardware Monitor to run". After all, the whole point is that both application are set by the user to start automatically so the user won't have to open them manually

Thanks for all the work you're doing!!!

**Comment by akej74** [comment]
Note on the two points above:

1. It's not trivial to do this from Python, i.e. access and change Windows Task Scheduler, I will postpone this idea for now. I will see if it's doable in some way, but it will have to wait for now.

2. Added a new option, "Start silently" (checkbox) that if enabled suppresses the OHM warning. Also changed the OHM warning message as per you suggestion above. 

Please try this out (release 1.0.8) and let me know if it works OK.

**Comment by ghost** [comment]
OK. I can confirm that the new OHM warning message has been applied. Thanks for that. Looks much better.

Also, something I've noticed. If you set the Grid Control application to start at log on by setting up a task in the Task Scheduler, the OHM application will start well before the Grid Control application and therefore the fix that you've created (including the on-screen warning message and the "Start silently" check box) become irrelevant.
However, if you set the Grid Control application to start at log on by placing a shortcut to the application's executable file in the Startup folder, then the Grid Control application starts BEFORE the OHM application and your fix is required.

My point is, since you've already created this fix then maybe, instead of accessing and changing Windows Task Scheduler you could just get the application to create a shortcut to the executable in the Startup folder. 

Not sure how hard that would be though. Just a thought. 

Thank you!!!

**Comment by akej74** [comment]
Thanks for your feedback, I will have a look into creating a shortcut in the Startup folder.

**Comment by CapitalF** [comment]
"Start silently" is somewhat ambiguous in it's meaning. Does this mean that the fans will start in silent mode? Until I read this bug report I had real doubts about what that checkbox did.

Regarding the Startup folder; this kind of application, which affects hardware, should probably be registered in the registry as a system autorun, which is easier to do pragmatically anyway.

I would suggest to do exactly like OHM does.

**Comment by akej74** [comment]
Hi, I agree that "Start silently" is not the best wording, let me think of something better. Maybe I could add a small "?" help icon for each checkbox, that displays a brief help text at mouse-over. 

Let me also look into the registration as an autorun, seems like the best way forward.
